packages/mobx-state-tree/src/utils.ts

```diff
@@ -178,3 +178,8 @@ deprecated = function(id: string, message: string): void {
 }
 deprecated.ids = {}
 export { deprecated }
+
+const GeneratorType = function*(): IterableIterator<any> {}.constructor
+export function isGenerator(fn: Function): boolean {
+    return fn instanceof GeneratorType
```

Comment:
Doesn't seem to work, every function (action) is detected as a generator

Reply:
Because this is how it is transpiled:
```
function () { return __generator(this, function (_a) {
    return [2 /*return*/];
}); }.constructor
```